### PR TITLE
complete assessment

### PR DESCRIPTION
**Pull request by DavonKenn**
# Submitting Week 1 Assessment

### ✅ Completed Files
Select the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // ASSESSMENT 2: Coding Practical Questions with Jest
 
-const { ifStatement, exportDefaultSpecifier } = require("@babel/types")
-const { describe } = require("yargs")
+// const { ifStatement, exportDefaultSpecifier } = require("@babel/types")
```
Comment (SunkissedQueen) [inline code-challenges.test.js:3]: Great. You remembered to comment out the "yargs"

File: code-challenges.test.js [hunk 2]
```diff
@@ -32,29 +32,49 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
-    } else ("num3" / 3 === NaN")
-}
+   const numbs = (number) => {
+     if(number %3 ===0) {
+       return `${number} is divisible by three`
+     } else if(number %3 !==0){
+       return `${number} is not divisible by three`
+     }
+   }
+console.log(numbs);
```
Comment (SunkissedQueen) [inline code-challenges.test.js:66]: If you want to see the function call. You have to give it an argument. Did your test pass?

File: code-challenges.test.js [hunk 3]
```diff
@@ -32,29 +32,49 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+
+
+
+
+// Psuedo Code
```
Comment (SunkissedQueen) [inline code-challenges.test.js:71]: Good pseudocode.

File: code-challenges.test.js [hunk 4]
```diff
@@ -64,67 +84,97 @@ const num3 = -7
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("nouns", () => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:87]: Good work

File: code-challenges.test.js [hunk 5]
```diff
@@ -64,67 +84,97 @@ const num3 = -7
...
 // b) Create the function that makes the test pass.
- {
-    if (randomNouns1.length) {
-      return [randomNouns1[0].toUpperCase()];
-    }
-    let res = capitalizeWords(randomNouns2.slice(0));
-    res.push(randomNouns2.slice(randomNouns2.length)[0].toUpperCase());
-    return res;
-   
+const nouns = (array) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:112]: Please checkout the basic structure of a function and then start adding the logic.

File: code-challenges.test.js [hunk 6]
```diff
@@ -64,67 +84,97 @@ const num3 = -7
...
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
  
 
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
+ describe(vowels,() => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:130]: You are doing better with the structure of your jest. High five! Place some quotation marks around the name of that function and then the jest suite should recognize your test.

File: code-challenges.test.js [hunk 7]
```diff
@@ -64,67 +84,97 @@ const num3 = -7
...
+    //     151 |    
+    //     152 | for (const cur of structuredClone.toLowerCase() {
+  
+
+//  Create a function that takes in a string and returns the index of the first vowel
+// identify what method i need to use to find the first vowel
+// find count of vowels
+// return number of vowels
+
+function countVowel(str) {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:168]: Did this work for you?

File: code-challenges.test.js [hunk 8]
```diff
@@ -64,67 +84,97 @@ const num3 = -7
...
+  
+
+//  Create a function that takes in a string and returns the index of the first vowel
+// identify what method i need to use to find the first vowel
+// find count of vowels
+// return number of vowels
+
+function countVowel(str) { 
+
+  const count = str.match(/[aeiou]/gi).length;
```
Comment (SunkissedQueen) [inline code-challenges.test.js:170]: Great work finding a new tool

File: coding-concepts.js [hunk 9]
```diff
@@ -7,22 +7,22 @@
 
 // --------------------1) What will this log?
 
-var cohort = "Bravo 2022"
+// var cohort = "Bravo 2022"
 //console.log(cohort.split(""))
 
 // a) Your answer: "B r a v o  2 0 2 2 "
-// b) Verify and explain:
+// b) Verify and explain: When you console log cohort.split with parenthesis it takes the values in the string puts them individually in a string
```
Comment (SunkissedQueen) [inline coding-concepts.js:14]: Yes and changes the datatype

File: coding-concepts.js [hunk 10]
```diff
@@ -7,22 +7,22 @@
...
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-//console.log(greeter("LEARN Student"))
+// console.log(greeter("LEARN Student"))
 
-// a) Your answer:It will log Hello name
-// b) Verify and explain: Ot came back undefined beacuse the function didnt have a specific name but the console log does.
+// a) Your answer:It will log Hello Learn Students
+// b) Verify and explain: It returned undefined because it doesnt have a return.
```
Comment (SunkissedQueen) [inline coding-concepts.js:25]: that's right

File: coding-concepts.js [hunk 11]
```diff
@@ -31,16 +31,16 @@ var multBy2 = [4, 5, 6, 7, 8].map(value => value * 2)
 //console.log(multBy2)
 
 // a) Your answer:[8,10,12,14,16]
```
Comment (SunkissedQueen) [inline coding-concepts.js:33]: good

File: coding-concepts.js [hunk 12]
```diff
@@ -54,7 +54,7 @@ var myCodingSkills = {
 //console.log(myCodingSkills.languages[0])
 
 // a) Your answer:JavaScript
```
Comment (SunkissedQueen) [inline coding-concepts.js:56]: great

File: coding-concepts.js [hunk 13]
```diff
@@ -67,7 +67,7 @@ class Learn {
   }
 }
 var learnStudent = new Learn("George")
-  console.log(learnStudent)
+  // console.log(learnStudent)
 
 // a) Your answer: this.student=George
```
Comment (SunkissedQueen) [inline coding-concepts.js:72]: Yes

File: interview-questions.md [hunk 14]
```diff
@@ -10,23 +10,23 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Your answer: A parameter goes next to a variable function and a argument goes with an if statement.
 
-  Researched answer: Parameters are properties of a function. Arguments are properties of a particular call to a function.
+  Researched answer: Parameters are properties of a function. Arguments are properties of a particular call to a function.You would use parameters as placeholders for an arugument in a function.A argument has to be a datatype recoginized by javascript and that data type is passed inside the parenthesis of a function invocation. The parameter and argument are different because the argument is using the parameter () to create an argument in console log.
```
Comment (SunkissedQueen) [inline interview-questions.md:13]: This answer is better. Thanks.

File: interview-questions.md [hunk 15]
```diff
@@ -10,23 +10,23 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer: Key values amd arrays. Both are optional.
+  Your answer: The built in methods are key values and arrays. Both are optional.
 
-  Researched answer:.map() have three built-in parameters - the value, index, and array. The index and array are both optional.
+  Researched answer: .map() is a higherorder function that has three built predefined parameters which are the value, index, and array. The index and array are both optional. The value is always required.
```
Comment (SunkissedQueen) [inline interview-questions.md:21]: When you have questions that can be answered quickly, fill the space with how you used the topic.

File: interview-questions.md [hunk 16]
```diff
@@ -10,23 +10,23 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What is the difference between map and filter?
 
   Your answer: Map is returning a new element to a new array. Filter is the same as Map but  returns the index. 
 
-  Researched answer:The .map() function iterates through an array and returns a new array of the same length.The .filter() function loops through an array and returns a new array with only the values that are truthy or the values that satisfy your stated condition.
+  Researched answer:The .map() function iterates through an array and returns a new array of the same length. The .filter() function loops through an array and returns a new array with only the values that are truthy or the values that satisfy your stated condition. Filter returns subset of the given array.
```
Comment (SunkissedQueen) [inline interview-questions.md:29]: You are showing you understand the research. Now tell the interviewer how you used that topic.

File: interview-questions.md [hunk 17]
```diff
@@ -35,7 +35,7 @@ Then, research the question to expand on your answer. Even if you feel you have
   Your answer: A function is telling a computer how to do something.A method is deals with objects in the function.
 
 
-  Researched answer: The difference is that a method is associated with an object, while a function is not.
+  Researched answer: The difference is that a method is associated with an object, while a function is not.methods ar functions stored as object properties. A function takes on a task,takes in value,takes in an input and returns an output.
```
Comment (SunkissedQueen) [inline interview-questions.md:38]: During research try to find something beyond the definition.  Maybe a unique way that the topic is being used.

File: interview-questions.md [hunk 18]
```diff
@@ -44,15 +44,15 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Your answer: Object destructuring is like breaking variables down or being more specific.
```
Comment (SunkissedQueen) [inline interview-questions.md:45]: I like this breakdown. If you support it with and example of how you used it in class..you will on the start of a great interview.

File: interview-questions.md [hunk 19]
```diff
@@ -44,15 +44,15 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Researched answer: Object destructuring allows you to take something like an array or an object and unpack their values into individual variables. 
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
   Your answer: Hoisting in javascript is moving code up.
 
-  Researched answer: Hoisting is moving declarations to the top.
+  Researched answer: Hoisting is functions or variable used before declarations.
```
Comment (SunkissedQueen) [inline interview-questions.md:55]: Cool. You answered the stretch. Give it the same approach. Define it, research something interesting, explain how you used it.

**Review by SunkissedQueen — APPROVED** [review]
Davon, your jest tests are looking so good. Great job! You also nailed the concepts. On the next assessment, 
1. Let's get a better handle on the functions you create to make your jest tests pass.
2. Interview questions - Try this technique:
a) Define the topic
b) Explain how you used it
c) Research something interesting about the topic (if you didn't define it correctly, explain your error)

File: code-challenges.test.js (hunk 2 above)
Comment (DavonKenn) [reply code-challenges.test.js:66]: I believe my test passed for this one.

**Pull request merged**